src/system/NewDialog/NewDialog.js

```diff
@@ -30,15 +31,20 @@ export const NewDialog = ( {
 	open = undefined,
 	id = undefined,
 } ) => {
+	const [ isOpen, setIsOpen ] = React.useState( defaultOpen );
+
 	if ( disabled ) {
 		return;
 	}
 
+	// if content is a function, pass in onClose
+	const isContentFunction = typeof content === 'function';
+
 	return (
 		<DialogPrimitive.Root
 			id={ id }
-			open={ open }
-			onOpenChange={ onOpenChange }
+			open={ open || isOpen }
```

Comment:
Can this be problematic? If the initial prop value for `open` is `true` and doesn't change, than the dialog will always stay open?

Reply:
@brunobasto You would need to customize the `onOpenChange`, yes. Do you have a suggestion for this case?